src/helpers/node/blocklists.js

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2021 RethinkDNS and its authors.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+import * as fs from "fs";
+import * as path from "path";
+import * as util from "../util.js";
+import * as envutil from "../envutil.js";
+
+const blocklistsDir = "./blocklists__";
```

Comment:
Can this be a named better?  `.cache/blocklists/` ? Or why not put it in `/tmp/` ?

If `.cache` is used, it needs to be added to `.gitignore` ?

Also, can the download start after the server's are up? Otherwise, considering download times are unpredictable, this also requires changes to tcp health checks on `fly.toml`, increase in grace period.

Reply:
> Can this be a named better? .cache/blocklists/ ? Or why not put it in /tmp/ ?

Deploying to vms managed by fly, even '/' is `/tmp`. Besides, [`/tmp`](https://en.wikipedia.org/wiki/Tmpfs) is sometimes mounted on the RAM... On ephemeral vms,  everything is fair-game in `/home`. And in local-dev, the files are right there in `blocklists__` folder of the project root.

> If .cache is used, it needs to be added to .gitignore 

Santhosh develops on Windows... `.cache` has no special meaning there. I prefer flat out creating the folder at project root where it is in-the-face so to speak... It _should_ be added to `.gitignore`, yes.

> Also, can the download start after the server's are up?

The idea is to set everything up where possible before servers are up. Including blocklists. Doesn't make sense to penalize the first few requests on fly where we've got a disk (same goes for local deploys).

> Otherwise, considering download times are unpredictable, this also requires changes to tcp health checks on fly.toml; increase in grace period to (?).

Yes, grace would probably need to be upped to `30s` or so. Good catch.

Reply:
> on Windows... .cache has no special meaning there

Yeah not Windows but software / tools may treat them different, like `eslint` ignores dot files/folders.

> Doesn't make sense to penalize the first few requests on fly where we've got a disk

On fly, downloads are fast. Maybe just causing a 1.5 sec latency on first request.

> grace would probably need to be upped to 30s or so

That's too much. 2s / 3s max.

Reply:
> Yeah, not Windows. But software / tools may treat them different, like eslint ignores dot files/folders by default.

`.cache`, while it make sense, complicates the `mkdir` logic (have to create `.cache`, then `blocklists`, then `timestamp` dirs; needless namespacing when there's nothing else in `.cache` except `blocklists`). I don't see any real value-add. bundle-time tools such as `eslint` et al may choose to ignore `.dotdirs` but that's orthogonal to the current discussion.

> That's too much. 2s / 3s; 5s max.

I don't think it is too much? Why do you say so? What other affects do you see happen due to, say, `60s` timeout.

Reply:
`.cache` should be enough...

> What other affects do you see happen due to, say, a 60s timeout.

Just the needless delay in delay deployment, fly deployments are the slowest of all 3 platforms already.

Reply:
> have to create .cache, then blocklists, then timestamp dirs

There is a [recursive option](https://nodejs.org/api/fs.html#fs_fs_mkdirsync_path_options):

```js
fs.mkdirSync("./.cache/blocklists/timestamp", { recursive: true });
```

Reply:
> fs.mkdirSync("./.cache/blocklists/timestamp", { recursive: true });

Neat.

I see some build systems, filesystems use `.cache` dirs. I'd suggest we leave it with `blocklists__` for now. It works, and there's no real benefit to renaming / moving it, either.

Reply:
> Just the needless delay in deployment, fly deployments are the slowest of all 3 platforms already.

Hm, we see 15s+ for downloads on Cloudflare at times, with 1.2s at the median. I believe 30s is "safe" since that was also once the upper limit on Cloudflare Workers (it is `15s` now, but with retries).